src/interactions/clickInteraction.ts

```diff
@@ -3,41 +3,58 @@
 module Plottable {
 export module Interaction {
   export class Click extends AbstractInteraction {
-    private _callback: (p: Point) => any;
+
+    private _mouseDispatcher: Plottable.Dispatcher.Mouse;
+    private _touchDispatcher: Plottable.Dispatcher.Touch;
+    private _clickCallback: (p: Point) => any;
+    private _clickedDown = false;
 
     public _anchor(component: Component.AbstractComponent, hitBox: D3.Selection) {
       super._anchor(component, hitBox);
-      hitBox.on(this._listenTo(), () => {
-        var xy = d3.mouse(hitBox.node());
-        var x = xy[0];
-        var y = xy[1];
-        this._callback({x: x, y: y});
-      });
+
+      this._mouseDispatcher = Dispatcher.Mouse.getDispatcher(<SVGElement> component.content().node());
+      this._mouseDispatcher.onMouseDown("Interaction.Click" + this.getID(), (p: Point) => this._handleMouseDown(p));
+      this._mouseDispatcher.onMouseUp("Interaction.Click" + this.getID(), (p: Point) => this._handleMouseUp(p));
+
+      this._touchDispatcher = Dispatcher.Touch.getDispatcher(<SVGElement> component.content().node());
+      this._touchDispatcher.onTouchStart("Interaction.Click" + this.getID(), (p: Point) => this._handleMouseDown(p));
+      this._touchDispatcher.onTouchEnd("Interaction.Click" + this.getID(), (p: Point) => this._handleMouseUp(p));
     }
 
-    public _requiresHitbox() {
-      return true;
+    private _handleMouseDown(p: Point) {
+      if (this._isInsideComponent(p)) {
+        this._clickedDown = true;
+      }
     }
 
-    protected _listenTo(): string {
-      return "click";
+    private _handleMouseUp(p: Point) {
+      if (this._clickedDown && this._isInsideComponent(p) && (this.onClick() != null)) {
```

Comment:
I remember a statement a while back that we should be using getters wherever possible.  Should we stick to that or?